lib/read.js

```diff
@@ -859,13 +855,19 @@ Reader.prototype.fromXML = function(xml, options, done) {
 
     var element = rootHandler.element;
 
-    // handle the situation that we could not extract
-    // the desired root element from the document
     if (!err && !element) {
       err = error('failed to parse document as <' + rootHandler.type.$descriptor.name + '>');
     }
 
-    done(err, err ? undefined : element, context);
+    if (err) {
+      err.parseContext = context;
+      reject(err);
```

Comment:
Please do `return` along with resolving / rejecting:

```suggestion
      return reject(err);
```